File: sourcePY/289.py
def gameOfLife(self, board):
    row = len(board)
    col = len(board[0]) if row > 0 else 0
    dx = [-1, -1, -1, 0, 0, 1, 1, 1]
    dy = [-1, 0, 1, -1, 1, -1, 0, 1]
    for i in range(row):
        for j in range(col):
            count = 0
            for k in range(8):
                ix, jy = i + dx[k], j + dy[k]
                if ix < 0 or ix >= row or jy < 0 or jy >= col:
                    continue
                if board[ix][jy] == 1 or board[ix][jy] == 2:
                    count += 1
            if not board[i][j] and count == 3:
                board[i][j] = 3
            if board[i][j] and (count < 2 or count > 3):
                board[i][j] = 2
    for i in range(row):
        for j in range(col):
            board[i][j] = board[i][j] % 2

File: source/289.cpp
int countLiveNeighbors(vector<vector<int>> board, int i, int j) {
	int res = 0, next = board[i][j];
	if (board[i - 1][j - 1]) res++;
	if (board[i - 1][j]) res++;
	if (board[i - 1][j + 1]) res++;
	if (board[i][j - 1]) res++;
	if (board[i][j + 1]) res++;
	if (board[i + 1][j - 1]) res++;
	if (board[i + 1][j]) res++;
	if (board[i + 1][j + 1]) res++;
	if (next && (res < 2 || res > 3)) next = 0;
	if (!next && res == 3) next = 1;
	return next;
}
void gameOfLife(vector<vector<int>>& board) {
	int row = board.size();
	if (row == 0) return;
	int col = board[0].size();
	if (col == 0) return;
	vector<vector<int>> copy(row + 2, vector<int>(col + 2, 0));
	for (int i = 0; i < row; ++i) {
		for (int j = 0; j < col; ++j) {
			copy[i + 1][j + 1] = board[i][j];
		}
	}
	for (int i = 0; i < row; ++i) {
		for (int j = 0; j < col; ++j) {
			board[i][j] = countLiveNeighbors(copy, i + 1, j + 1);
		}
	}
}